### PR TITLE
Add substring from to char

### PR DESCRIPTION
**Pull request by krwq**

### CODE REVIEW

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs [hunk 1]
```diff
@@ -337,16 +337,117 @@ public Utf8String Substring(int index, int length)
...
+            return -1;
+        }
+
+        // TODO: Should this be public or should focused on scenarios?
+        private int IndexOf(UnicodeCodePoint codePoint)
         {
             throw new NotImplementedException();
         }
 
+        public bool TrySubstringFrom(Utf8String substring, out Utf8String result)
```
Comment (KrzysztofCwalina) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: The first parameter should not be called substring.

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs [hunk 2]
```diff
@@ -337,16 +337,117 @@ public Utf8String Substring(int index, int length)
...
+            {
+                result = default(Utf8String);
+                return false;
+            }
+
+            result = Substring(idx);
+            return true;
+        }
+
+        public bool TrySubstringFrom(Utf8CodeUnit codeUnit, out Utf8String result)
```
Comment (KrzysztofCwalina) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: It's also not clear and possibly needs to be controllable where the delimiter ends up being: in the result or in the rest.

Comment (krwq) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: From sounds inclusive to me. Let's discuss naming on the API review

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs (hunk 1 above)
Comment (krwq) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: Well, it is a substring or the method returns false, what name do you propose?

Comment (KrzysztofCwalina) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: To me the result is the substring. The substring parameter is more like a prefix the method will be searching for. But please feel to commit this change, just make a note so we talk about this name at the API review.

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs (hunk 2 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: Yeah, let's discuss separately.

**Comment by KrzysztofCwalina** [comment]
Squash the commits, and it's good to go.

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs (hunk 1 above)
Comment (krwq) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: Renaming to "value" as in string.IndexOf

**Pull request merged**